todo.org

```diff
@@ -0,0 +1,80 @@
+* Core
+
+** TODO set repl type on connection not first command
+For some reason ~inf-clojure--set-repl-type~ is called in:
+1. inf-clojure--send-string
+2. inf-clojure-reload-form
+3. inf-clojure-reload-all-form
+
+Seems better to do this on the two different connection methods and then be done with it?
+
+** DONE do we need repl type in both source buffer and connection?
+   these can get out of sync and lead to confusing errors when closing a repl and opening a new one. It seems like we keep the repl-type in the source buffer to prevent a single ~(with-current-buffer (process-buffer proc) inf-clojure-repl-type)~
+
+** TODO nice startup
+There's some project detection but that's becoming less and less useful as time goes on. Shadow, lein, deps.edn can all easily be mixed in the same project. And then lumo, planck, or bb scripts could live side by side. Rather than trying to guess the project type, I think i'd like to mimic geiser's style of handling multiple scheme backends. Perhaps ~m-x inf-clojure-run-planck~ and similar could help out.
```

Comment:
I would say that project detection is very tricky cause for instance lumo files could still use a project.clj file for generating the classpath.

I would actually start considering static file/buffer repl detection (instead of find-ns we identify namespaces in the file content)

Reply:
yeah these are just sketches for a nicer startup. and its for future work :)